lms/djangoapps/instructor/views/api.py

```diff
@@ -1717,6 +1721,61 @@ def get_student_enrollment_status(request, course_id):
     return JsonResponse(response_payload)
 
 
+class StudentProgressUrlSerializer(serializers.Serializer):
+    """Serializer for course renders"""
+    unique_student_identifier = serializers.CharField(write_only=True)
+    course_id = CourseKeyField(required=False)
+    progress_url = serializers.SerializerMethodField()
+
+    def get_progress_url(self, obj):    # pylint: disable=unused-argument
+        """
+        Return the progress URL for the student.
+
+        Args:
+            obj (dict): The dictionary containing data for the serializer.
+
+        Returns:
+            str: The URL for the progress of the student in the course.
+        """
+        user = get_student_from_identifier(obj.get('unique_student_identifier'))
+        course_id = obj.get('course_id')  # Adjust based on your data structure
+
+        if course_home_mfe_progress_tab_is_active(course_id):
+            progress_url = get_learning_mfe_home_url(course_id, url_fragment='progress')
+            if user is not None:
+                progress_url += '/{}/'.format(user.id)
+        else:
+            progress_url = reverse('student_progress', kwargs={'course_id': str(course_id), 'student_id': user.id})
+
+        return progress_url
+
+
+@method_decorator(cache_control(no_cache=True, no_store=True, must_revalidate=True), name='dispatch')
+class StudentProgressUrl(APIView):
+    """
+    Get the progress url of a student.
+    Limited to staff access.
+    Takes query parameter unique_student_identifier and if the student exists
+    returns e.g. {
+        'progress_url': '/../...'
+    }
+    """
+    permission_classes = [IsAuthenticated, permissions.InstructorPermission]
```

Comment:
We should add appropriate authentication class

Reply:
I simply matched the old behaviour.